### PR TITLE
Feature: add support for creating projects through the python api

### PR DESCRIPTION
**Pull request by axl1313**
### Description
Adds support for creating projects through they Python API. User must supply dataset_id and project_name parameters. All other project fields are specified through optional kwargs. For fields that aren't supplied by the user, we make our best guess (similar to how the frontend supplies defaults)

### How to Test
- Test with backend changes in feat/python-api-project-creation
- Test that creating projects works for all modalities
- Try various combinations of user supplied and inferred project fields

### CODE REVIEW

File: cleanlab_studio/studio/studio.py
Comment (ryansingman) [inline cleanlab_studio/studio/studio.py:164]: thought: we want to deprecate dataset modality in the near future, and make modality a project-only construct. because of this, I don't think we want to make modality optional.

File: cleanlab_studio/studio/studio.py
Comment (ryansingman) [inline cleanlab_studio/studio/studio.py:158]: since this is a fairly complex interface (with some non-obvious arguments) and is user-facing, we definitely need to add a docstring here. docstring should focus on the high-level meaning of each argument, from a user perspective

File: cleanlab_studio/studio/studio.py
Comment (ryansingman) [inline cleanlab_studio/studio/studio.py:180]: nit: "Using all valid feature columns"

File: cleanlab_studio/studio/studio.py
Comment (ryansingman) [inline cleanlab_studio/studio/studio.py:176]: as a general principle, we should be failing on these types of things. if a user supplies an argument, they expect it to be used. if the argument can't be used, we should fail and alert the user. same principle applies for the text column conditional below

**Review by ryansingman — CHANGES_REQUESTED** [review]
Mostly LGTM, some points to resolve before merging thouggh

File: cleanlab_studio/studio/studio.py
Comment (axl1313) [reply cleanlab_studio/studio/studio.py:164]: Ah good call. That makes sense to me

File: cleanlab_studio/studio/studio.py
Comment (ryansingman) [inline cleanlab_studio/studio/studio.py:155]: suggestion: let's type these as literals instead of strings, to make clearer to the user what the choices are

**Review by ryansingman — APPROVED** [review]
One suggested improvement, otherwise LGTM. If you do make the change I suggested, no need for another review -- just go ahead and merge

**Pull request merged**
